### PR TITLE
[packaging] Add `--no-deps` option to ignore dependency conflicts when bootstrapping

### PR DESCRIPTION
**Pull request by jjotero**

### CODE REVIEW

**Review by teojgo — APPROVED** [review]

**Review by victorusu — APPROVED** [review]
`lgtm`

File: bootstrap.sh [hunk 1]
```diff
@@ -35,6 +35,7 @@ usage()
     echo "  -h       Print this help message and exit"
     echo "  +docs    Build also the documentation"
     echo "  +pygelf  Install also the pygelf Python package"
+    echo "  +nodeps  Ignore incompatible Python package dependencies"
```
Comment (vkarak) [inline bootstrap.sh:38]: This should be starting with `-`. The `+` is for features to enable.

**Review by vkarak — CHANGES_REQUESTED** [review]

File: bootstrap.sh (hunk 1 above)
Comment (victorusu) [reply bootstrap.sh:38]: Great catch!

Comment (jjotero) [reply bootstrap.sh]: Uops! Done!

File: bootstrap.sh [hunk 2]
```diff
@@ -35,6 +35,7 @@ usage()
     echo "  -h       Print this help message and exit"
     echo "  +docs    Build also the documentation"
     echo "  +pygelf  Install also the pygelf Python package"
+    echo "  -nodeps  Ignore incompatible Python package dependencies"
```
Comment (vkarak) [inline bootstrap.sh:38]: I would use `--no-deps` for POSIX compatibility and move it up in the help message.

**Review by vkarak — CHANGES_REQUESTED** [review]

**Comment by vkarak** [comment]
What about instead of `--no-deps` adding a `--pip-opts` option that would pass through any option to `pip`?

**Review by vkarak — APPROVED** [review]

**Pull request merged**